### PR TITLE
New updates

### PR DESCRIPTION
**Pull request by CalebUAz**
1. Added a slide with the text "Don't discuss "before the first image.
2. After the image shows up and the timer hits zero, then add another slide with the text "Now you may discuss".

### CODE REVIEW

File: tasks/affective_task/client_affective_task.py [hunk 1]
```diff
@@ -54,11 +57,21 @@ def run(self, collaboration: bool = False):
             wait(CROSS_SCREEN_MILLISECONDS)
             
             if collaboration:
+                # displaying a slide asking subjects not to discuss 
+                render_blank_screen(self._screen, BLANK_SCREEN_MILLISECONDS)
+                display_msg_affective_disscussion(self._screen, "Observe!",DISPLAY_AFFEC_DISCUSSION_MILLISECONDS/2)
```
Comment (eduongAZ) [inline tasks/affective_task/client_affective_task.py:62]: Remove the exclamation mark, keep the message as "Observe", not "Observe!"

File: tasks/affective_task/client_affective_task.py [hunk 2]
```diff
@@ -54,11 +57,21 @@ def run(self, collaboration: bool = False):
...
+
                 # show an image for team task for the team to analyze seperately
                 render_image_center(state["image_path"], self._screen, refresh=True)
                 render_text_center("Quiet", (950, 50), self._screen, font_size = 45 , x_offset = 0, y_offset=450)
                 timer(state["image_timer"], [], "Team: " if collaboration else "Individual: ", self._screen)
-                
+
+                # displaying a slide asking subjects to discuss 
+                render_blank_screen(self._screen, BLANK_SCREEN_MILLISECONDS)
+                display_msg_affective_disscussion(self._screen, "Discuss!",DISPLAY_AFFEC_DISCUSSION_MILLISECONDS/2)
```
Comment (eduongAZ) [inline tasks/affective_task/client_affective_task.py:72]: Remove the exclamation mark, keep the message as "Discuss", not "Discuss!"

File: tasks/affective_task/client_affective_task.py [hunk 3]
```diff
@@ -202,8 +224,11 @@ def button_response(events) -> bool:
 
                     return False
 
-
-            timer(state["rating_timer"], [button_response], "Team: " if collaboration else "Individual: ", self._screen)
+            if collaboration:
+                timer(state["rating_timer"], [button_response], "Warning! Rating ends in: ", self._screen, display_timer = False)
```
Comment (eduongAZ) [inline tasks/affective_task/client_affective_task.py:228]: Just "Rating ends in: " is good, the "Warning!" is too exclamatory/

File: tasks/affective_task/utils/__init__.py [hunk 4]
```diff
@@ -4,3 +4,4 @@
 from common import render_text_center
 from .submit_button import submit_button
 from .timer import timer
+from .display_msg_affective_disscussion import display_msg_affective_disscussion
```
Comment (eduongAZ) [inline tasks/affective_task/utils/__init__.py:7]: Add a new line here

File: tasks/affective_task/utils/display_msg_affective_disscussion.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+from common import render_text_center
+
+def display_msg_affective_disscussion(screen ,msg: str,milliseconds: int):
+    start_ticks = pygame.time.get_ticks()
+
+    clock = pygame.time.Clock()
+    while pygame.time.get_ticks() - start_ticks < milliseconds:
+        render_text_center(msg, (1250, 90), screen, font_size = 55 , x_offset = 0, y_offset=0)
+        pygame.event.get()
+        clock.tick(UPDATE_RATE)
```
Comment (eduongAZ) [inline tasks/affective_task/utils/display_msg_affective_disscussion.py:12]: Add a new line here

File: tasks/affective_task/utils/timer.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 REFRESH_RATE = 20
 
 
-def timer(seconds: int, callbacks: List[Callable], pre_text: str, screen):
+def timer(seconds: int, callbacks: List[Callable], pre_text: str, screen,display_timer = True):
```
Comment (eduongAZ) [inline tasks/affective_task/utils/timer.py:12]: This feels misleading: if the user sets False to display_timer, the user expects that the screen does not show the timer at all, but here the timer is displayed after the time is halved.

You should consider having the argument "display_mode" instead, with 0 being show no timer, 1 is show timer, and 2 is show timer half the time.

File: tasks/affective_task/utils/display_msg_affective_disscussion.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import pygame
+from config import UPDATE_RATE
+from common import render_text_center
+
+def display_msg_affective_disscussion(screen ,msg: str,milliseconds: int):
```
Comment (eduongAZ) [inline tasks/affective_task/utils/display_msg_affective_disscussion.py:5]: space after the comma, not before.

 `def display_msg_affective_disscussion(screen, msg: str, milliseconds: int):`

File: tasks/affective_task/client_affective_task.py [hunk 8]
```diff
@@ -202,8 +224,11 @@ def button_response(events) -> bool:
...
+            else:
+                timer(state["rating_timer"], [button_response], "Individual: ", self._screen)
+            #timer(state["rating_timer"], [button_response], "Team: " if collaboration else "Individual: ", self._screen)
```
Comment (eduongAZ) [inline tasks/affective_task/client_affective_task.py:231]: Remove commented code.

**Review by eduongAZ — CHANGES_REQUESTED** [review]
Delete all .DS_Store files

File: tasks/affective_task/utils/timer.py (hunk 6 above)
Comment (CalebUAz) [reply tasks/affective_task/utils/timer.py]: Done

**Review by eduongAZ — CHANGES_REQUESTED** [review]
There might be an out-of-sync bug

File: tasks/affective_task/client_affective_task.py [hunk 9]
```diff
@@ -69,7 +82,16 @@ def run(self, collaboration: bool = False):
...
                 timer(state["image_timer"], [], "Team: " if collaboration else "Individual: ", self._screen)
+
+            if collaboration and state["selected"]:
+                # slide before that shows up based on the client that is selected before the buttons are displayed
+                render_blank_screen(self._screen, BLANK_SCREEN_MILLISECONDS)
+                display_msg_affective_disscussion(self._screen, "You have been selected for rating the images",DISPLAY_AFFEC_DISCUSSION_MILLISECONDS)
+                render_blank_screen(self._screen, BLANK_SCREEN_MILLISECONDS)
             
+            else:
+                render_blank_screen(self._screen, BLANK_SCREEN_MILLISECONDS)
```
[the remark this answers is not in the document]
Comment (CalebUAz) [reply tasks/affective_task/client_affective_task.py:93]: Yeah, your right! I'll add that as well!

[the remark this answers is not in the document]
Comment (CalebUAz) [reply tasks/affective_task/client_affective_task.py]: Done!

**Review by eduongAZ — APPROVED** [review]

**Pull request merged**